### PR TITLE
Patched to be buildable on FreeBSD 13.0-RELEASE

### PR DESCRIPTION
**Pull request by SamuelGogarty**
## Changes Proposed:
-  patched Util.cpp, System.h and Filesystem.cpp to be libc compatible and buildable on FreeBSD
-  

## Issues Addressed:

- Closes 

## Tests Performed:

-  compiled and ran server
- connected to it with a client and leveled up a few times
- 

## How to Test the Changes:

1. Build on FreeBSD then connect

## Known Issues and TODO List:

Write documentation explaining the installation process on FreeBSD
- [ ]
- [ ]

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

**Comment by acidmanifesto** [comment]
I am not a FreeBSD person at all nor will I pretend to be one. 
Only thing i recommend outside of the testing is to verify time. 
Does game time progress normally? Is there a default time zone this sets it to?
Also is there any issues with the events or holidays triggering and durations?
Perfect test would be the hourly bell event.
.event start 73

https://github.com/azerothcore/azerothcore-wotlk/blob/22be8f105e7abcee5078e7c07c8742f51bf402ed/src/common/Utilities/Util.cpp#L92-L97

**Comment by SamuelGogarty** [comment]
> I am not a FreeBSD person at all nor will I pretend to be one. Only thing i recommend outside of the testing is to verify time. Does game time progress normally? Is there a default time zone this sets it to? Also is there any issues with the events or holidays triggering and durations? Perfect test would be the hourly bell event. .event start 73
> 
> https://github.com/azerothcore/azerothcore-wotlk/blob/22be8f105e7abcee5078e7c07c8742f51bf402ed/src/common/Utilities/Util.cpp#L92-L97

tested. works as intended

**Comment by acidmanifesto** [comment]
> > I am not a FreeBSD person at all nor will I pretend to be one. Only thing i recommend outside of the testing is to verify time. Does game time progress normally? Is there a default time zone this sets it to? Also is there any issues with the events or holidays triggering and durations? Perfect test would be the hourly bell event. .event start 73
> > https://github.com/azerothcore/azerothcore-wotlk/blob/22be8f105e7abcee5078e7c07c8742f51bf402ed/src/common/Utilities/Util.cpp#L92-L97
> 
> tested. works as intended

May the auto compile testing forever be in your favor. Thanks for the PR. Lets hope this passes.

**Comment by Kitzunu** [comment]
You are breaking windows and modules

**Comment by SamuelGogarty** [comment]
> You are breaking windows and modules

neither of these errors are caused by my patch.

**Comment by acidmanifesto** [comment]
> You are breaking windows and modules

Seems there was a error on the windows build not installing boost before it even does a compile run. Can we somehow rerun that check?
The modules seems unrelated as it mentions sql scripts not being attached? This pr doesnt touch that but maybe that test needs to be reran as well?

**Comment by Kitzunu** [comment]
Modules are broken on master
Windows rerunning

**Comment by SamuelGogarty** [comment]
> > You are breaking windows and modules
> 
> Seems there was a error on the windows build not installing boost before it even does a compile run. Can we somehow rerun that check? The modules seems unrelated as it mentions sql scripts not being attached? This pr doesnt touch that but maybe that test needs to be reran as well?

The modules is an already known issue unrelated to this.

`Shin — Today at 12:19 PM
thanks @Kaiser I set the run-build label so all build can run
the modules build will probably fail because we currently have a problem with a module`

**Comment by Kitzunu** [comment]
But now windows is actually broken

File: deps/g3dlite/include/G3D/System.h [hunk 1]
```diff
@@ -20,7 +20,7 @@
 #include "G3D/BinaryFormat.h"
 #include "G3D/FileNotFound.h"
 #include <string>
-
+#include <sys/time.h>
 #if defined(__aarch64__)
 #include <sys/time.h>
```
Comment (Kitzunu) [inline deps/g3dlite/include/G3D/System.h:25]: Breaks windows, thats why the if aarch64 is there.

**Comment by acidmanifesto** [comment]
@SamuelGogarty 
D:\a\azerothcore-wotlk\azerothcore-wotlk\deps\g3dlite\include\G3D/System.h(23,10): fatal error C1083: Cannot open include file: 'sys/time.h': No such file or directory

**Comment by SamuelGogarty** [comment]
ill look into it

File: deps/g3dlite/include/G3D/System.h (hunk 1 above)
Comment (SamuelGogarty) [reply deps/g3dlite/include/G3D/System.h:25]: It fails to build on fbsd if its done within the if, hm

Comment (Kitzunu) [reply deps/g3dlite/include/G3D/System.h:25]: Include if FreeBDS?

Comment (Kitzunu) [reply deps/g3dlite/include/G3D/System.h:25]:
```cpp
# if defined(__aarch64__) || defined(__FreeBSD__)
```

**Comment by SamuelGogarty** [comment]
@FrancescoBorzi @Kitzunu builds passed

**Comment by Kitzunu** [comment]
@Winfidonarleyan

**Review by Winfidonarleyan — APPROVED** [review]

**Pull request merged**